app/jobs/chat_job.rb

```diff
@@ -0,0 +1,295 @@
+class ChatJob < ApplicationJob
+  queue_as :default
+
+  def perform(chat_id, message_id)
+    chat = Chat.find(chat_id)
+    nil if chat.nil?
+
+    message = chat.messages.find(message_id)
+    openai_client = OpenAI::Client.new
+
+    viability = determine_viability(openai_client, chat)
+
+    if !viability["content_contains_answer"]
+      handle_non_viable_conversation(openai_client, chat, message, viability)
+    else
+      generate_response_content(openai_client, chat, message)
+    end
+  end
+
+  private
+    def determine_viability(openai_client, chat)
+      chat_history = chat.messages.where.not(content: [ nil, "" ]).where.not(content: "...").where.not(role: "log").order(:created_at)
```

Comment:
If you extract the scope for `where.not(content: "...")` then you can reuse that here.